.travis.yml

```diff
@@ -3,12 +3,30 @@ go:
   - 1.7
   - tip
 
+install:
+  - wget https://github.com/google/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip
+  - unzip protoc-3.1.0-linux-x86_64.zip -d /tmp
+  - sudo cp /tmp/bin/protoc /usr/bin/protoc
+  - sudo chmod 777 /usr/bin/protoc
+  - rm protoc-3.1.0-linux-x86_64.zip
+
 before_script:
-  - go get github.com/ChimeraCoder/gojson/gojson
+  - go get -u github.com/ChimeraCoder/gojson/gojson
+  - go get -u github.com/golang/protobuf/protoc-gen-go
   - go generate
+  # We need to ignore changes to this one file
+  # because the go-generated version will have a different
+  # fileDescriptor0 depending on which version of Go was used
+  # to build protoc-gen-go.
+  - git checkout ssf/sample.pb.go
+
   - gofmt -w .
-  - test -n $(git status --porcelain)
 
+  # If `go generate` or `gofmt` yielded any changes,
+  # this will fail with an error message like "too many arguments"
+  # or "M: binary operator expected"
```

Comment:
per our past discussion, we should probably put some more detail into this comment (ie about how git-update-index and git-diff-index interact)